### DISCUSSION TITLE
Truncate rebase conflict status commit list if too long

### DISCUSSION THREAD

**Discussion by greened** [magit/magit#4722] [Ideas]
When I run magit over TRAMP, do a very large rebase (such as rebasing downstream changes on top of a new upstream release) and encounter a conflict, magit spends a lot of time inserting buffers for each commit in the rebase sequence to display the status.  Since this is over TRAMP, each such insert requires network operations and it proceeds very slowly, to the point where I have to `quit` and lose the ability to operate on the rebase from within magit.

It would be helpful to have an option to limit the number of such commits magit tries to display in the rebase status.  I really only care about the ~10 commits surrounding the current conflicting commit.  I don't care about things 1000 commits in the future.  I think a simple customization option would be fine.  By default it would be "infinite" to maintain current behavior but could be set to some integer to limit the work magit does on each conflict.

**Comment by greened** [comment]
In addition, it would be nice if a `quit` during the "insert a buffer for each rebase commit" sequence would simply stop that activity but allow magit to continue rendering the other parts of the status (the file state/conflict diff) so that we can use normal magit operations to visit the files with conflicting changes.

**Comment by tarsius** [comment]
> "insert a buffer for each rebase commit"

Do you remember what you mean by that?

I don't think we "insert a buffer" for every commit when listing the commits yet to be rebased. What does "insert" mean here anyway? Maybe "create"?

I didn't spot anything where we create a buffer per pending commit. Maybe there used to be. I vaguely remember that there was some major inefficiency related to listing these commits, but don't remember if it was related to buffers, and cannot find the change.

I'm gonna assume this was fixed. If that's not correct, please tell me more about these buffers, so that I can determine where they are being created (or if you know, just tell me).
